metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -584,7 +584,17 @@ def generate_partition_profiler_query(
         if partition:
             partition_ts: Union[datetime.datetime, datetime.date]
             if not partition_datetime:
-                partition_datetime = parser.parse(partition.partition_id)
+                """
+                bigquery partition_id of the MONTHLY partition is a format of {year}{month} such as 202204.
+                dateutil.parser.parse consider six digits string as {day}{month}{year}.
+                Thus we need to append a seperator like '-' between 2022 and 04.
+                """
+                partition_id_for_parse = partition.partition_id
+                if len(partition_id_for_parse) == 6:
+                    partition_id_for_parse = (
```

Comment:
Can we do something like this which I think is much clearer?:
```
partition_datetime = datetime.strptime(partition_id_for_parse,"%Y%M")
```

Reply:
We need to use the 'dateutil.parser' anyway because we can have YEARLY / DAILY partition as well. @treff7es

Reply:
If we use 'datetime.strptime' only for monthly case, we have to use two parsing functions ('datetime.strptime' and 'dateutil.parser.parse') to cover all cases. I think we'd better to use one function for the consistency.
Also, if we use 'datetime.strptime', there will be a difference. For example, when we parse '202204' with strptime, the result will be '2022-04-01 00:00:00', and if we parse '2022-04' with 'dateutil.parser.parse' in 2022-04-21 (the actual date), the result will be '2022-04-21 00:00:00'. @treff7es